### DISCUSSION TITLE
Going to "floorp.app" will go to the russian page by default.

### DISCUSSION THREAD

**Discussion by AxeldeWater** [Floorp-Projects/Floorp#1138] [Feature Request]
I didn't know where to report this, so I will just do it here: whenever you open "floorp.app" without any language tags, it will default to russian. I don't think this should be the behaviour. It should be either English (because this is the Lingua Franca of the internet) or Japanese (because the browser is Japanese).

**Comment by regularentropy** [comment]
It depends on your locale. In my case, it defaults to English.
Please provide a GIF if possible.

**Comment by AxeldeWater** [comment]
![screen-20240510-115256.gif](https://github.com/Floorp-Projects/Floorp/assets/110288748/fa382db5-7f08-4c60-8f3c-8ee1b8ab9bd4)

I just noticed that this only happens in my chromium based browser, except Opera. Only Opera mobile and Firefox Mobile default to English, Google Chrome, Cromite and Bromite all default to the russian page, on my phone. I will test this all out on my desktop too. 

But even then, my default language is set to be English (and otherwise Dutch). I might have Russian in my language list too since I can speak Russian but it is not my default language. 

Edit: uhm for some reason the gif got corrupted? Very weird, give me a moment I will provide a new gif.

Edit 2: Here is the new gif

![GIF_20240510_121527_696.gif](https://github.com/Floorp-Projects/Floorp/assets/110288748/8e130116-e8d6-41d4-924a-11cce7fe72eb)

**Comment by creeper-0910** [comment]
The defaultLocale is set to en and I have not been able to reproduce that problem in my environment.

**Comment by AxeldeWater** [comment]
Hmm... I will check my language settings...

**Comment by AxeldeWater** [comment]
I found the reason. My language priority list was 1. English (United Kingdom) 2. Dutch 3. Russian 4. English 5. English (United States). The site for some reason does not recognize "English (United Kingdom)" as being an English locale so it defaults to Russian. I changed the order and now it is good. But still, the fact that English (United Kingdom) is not recognized as English is a problem. Maybe the same is true for the United States version and the other versions.

**Reply by creeper-0910** [reply]
Perhaps this is due to the fact that the en-gb translation is not yet implemented.

**Reply by AxeldeWater** [reply]
I would say that en-gb and en-us (and any other version) could (temporarily) be a copy of the normal English version. Because I don't think I am the only one that just has UK or US version as the default language instead of the normal English. Finding yourself on a page in a different language could be off-putting in such cases.

**Reply by regularentropy** [reply]
>Perhaps this is due to the fact that the en-gb translation is not yet implemented.

@creeper-0910 I'll implement it on this week

**Reply by creeper-0910** [reply]
It may be an Accept-Language issue.
Can you please provide information on the Accept-Language header?

**Reply by AxeldeWater** [reply]
@creeper-0910 was this question for me? How do I check this header? Or am I just confused now?

**Comment by AxeldeWater** [comment]
Confirmed this problem also for "English (United States)"

**Reply by joshika39** [reply]
Sorry for barging in here!

I want to ask you to wait a little, a refactored version of the website is being deployed, where all of these issues are addressed and fixed!

**Reply by joshika39** [reply]
More over it has the language header detection for all of the working translations (currently en, and ja)

If you want to contribute to the translations you can do it here: https://crowdin.com/project/floorp-web

**Update:** Forgot to mention, it detects all of the English languages so if you have any other locale of the english language. It will be properly redirected.

**Reply by creeper-0910** [reply]
Sorry, I forgot to reply.
As mentioned above, we are refactoring the website, so please be patient for a while!

**Comment by regularentropy** [comment]
Fixed: https://github.com/Floorp-Projects/web/pull/43

**Comment by AxeldeWater** [comment]
I can confirm it works now. Also: wow, that redesign! Looks great!!!
